deploy/crds/applicationmonitoring.integreatly.org_applicationmonitorings_crd.yaml

```diff
@@ -10,33 +10,83 @@ spec:
     plural: applicationmonitorings
     singular: applicationmonitoring
   scope: Namespaced
-  version: v1alpha1
+  subresources:
+    status: {}
```

Comment:
Setting status to be a subresource might actually be a bad idea in this case :thinking: Because of the other ways that this operator is broken someone may need to manually change the status phase to force the operator to re-install a component or run some other logic that is currently being skipped based on a CR being at the last phase.

I'll play with it for a while on a cluster to see if this is a valid concern or not.

Reply:
I was correct, it's a bad idea. When it's set as a subresource, while the K8s API server will "accept" changes to fields in the status from e.g. `oc edit`, it won't actually do anything with them. This, in combination with the fact that if it's at phase 9 it won't do anything from previous phases, means that if something that it creates (e.g. Grafana CR) is deleted, then there's no way to get it to re-create without uninstalling and reinstalling the entire monitoring stack.

I've changed it back to not being a subresource, and I've removed the annotation from the types file that causes it to be generated as such. Not being a subresource is also not great, because when it updates a field in the status, then the "generation" of the CR changes, and that's a lot because there's a field called `lastblackboxconfig` which stores the md5 hash of something for some reason. This is the underlying cause of https://issues.redhat.com/browse/INTLY-5018

In the words of Bart Simpson:
> Well, you're damned if you do, and you're damned if you don't.